### PR TITLE
WIP: Deprecate SbSocket and migrate to socket APIs

### PR DESCRIPTION
**Pull request by maxz-lab**
SbSocketClearLastError -> errno
SbSocketGetLastError   -> errno

b/302701164

Change-Id: Ie1c099743a8cec011c890c87cb0b23ec945ca1a0

### CODE REVIEW

File: starboard/socket.h [hunk 1]
```diff
@@ -126,6 +137,41 @@ typedef struct SbSocketResolution {
 // Well-defined value for an invalid socket handle.
 #define kSbSocketInvalid ((SbSocket)NULL)
 
+#if SB_API_VERSION >= 16
```
Comment (y4vor) [inline starboard/socket.h:140]: This code should go in starboard/common. 
We don't allow c++ code in the starboard/foo.h headers.

File: net/base/net_errors.h [hunk 2]
```diff
@@ -12,7 +12,11 @@
 #include "base/logging.h"
 #include "net/base/net_export.h"
 #if defined(STARBOARD)
+#if SB_API_VERSION < 16
 #include "starboard/common/socket.h"
+#else
+#include <errno.h>
```
Comment (y4vor) [inline net/base/net_errors.h:18]: Please file a bug to make <errno.h> work!

File: net/socket/tcp_socket_starboard.cc [hunk 3]
```diff
@@ -50,7 +55,32 @@ TCPSocketStarboard::~TCPSocketStarboard() {
 int TCPSocketStarboard::Open(AddressFamily family) {
   DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
   DCHECK(!SbSocketIsValid(socket_));
+#if SB_API_VERSION < 16
   socket_ = SbSocketCreate(ConvertAddressFamily(family), kSbSocketProtocolTcp);
+#else
+  int socket_fd = socket(ConvertSocketAddressTypeToDomain(ConvertAddressFamily(family)),
```
Comment (y4vor) [inline net/socket/tcp_socket_starboard.cc:61]: Is this just temporary till you get all the APIs adjusted?

File: net/socket/tcp_socket_starboard.cc [hunk 4]
```diff
@@ -50,7 +55,32 @@ TCPSocketStarboard::~TCPSocketStarboard() {
...
+    socket_= kSbSocketInvalid;
+  } else if (!SocketSetNonBlocking(socket_fd)) {
+    // Something went wrong, we'll clean up (preserving errno) and return
+    // failure.
+    int save_errno = errno;
+    HANDLE_EINTR_WRAPPER(close(socket_fd));
+    errno = save_errno;
+    socket_ = kSbSocketInvalid;
+  } else {
+    socket_ = new SbSocketWrapper(ConvertAddressFamily(family), kSbSocketProtocolTcp, socket_fd);
```
Comment (y4vor) [inline net/socket/tcp_socket_starboard.cc:73]: Why do you need a SbSocketWrapper?

File: starboard/elf_loader/exported_symbols.cc [hunk 5]
```diff
@@ -314,6 +314,8 @@ ExportedSymbols::ExportedSymbols() {
   REGISTER_SYMBOL(SbStringFormat);
   REGISTER_SYMBOL(SbStringFormatWide);
   REGISTER_SYMBOL(SbStringScan);
+  REGISTER_SYMBOL(SbSocketClearLastError);
```
Comment (y4vor) [inline starboard/elf_loader/exported_symbols.cc:317]: Why are you introducing new Starboard symbols in SB 16?

File: starboard/elf_loader/exported_symbols.cc [hunk 6]
```diff
@@ -268,13 +268,13 @@ ExportedSymbols::ExportedSymbols() {
 #endif  // SB_API_VERSION >= 15
   REGISTER_SYMBOL(SbSocketAccept);
   REGISTER_SYMBOL(SbSocketBind);
-  REGISTER_SYMBOL(SbSocketClearLastError);
   REGISTER_SYMBOL(SbSocketConnect);
+#if SB_API_VERSION < 16
   REGISTER_SYMBOL(SbSocketCreate);
```
Comment (y4vor) [inline starboard/elf_loader/exported_symbols.cc:273]: Where is the posix `socket` API registration?

File: starboard/common/socket.cc [hunk 7]
```diff
@@ -105,11 +106,17 @@ bool Socket::IsPending() {
 }
 
 SbSocketError Socket::GetLastError() {
```
[the remark this answers is not in the document]
Comment (maxz-lab) [reply starboard/common/socket.cc:253]: I would say NO, not going to carry the SbSocketError to SB16+.

File: net/socket/tcp_socket_starboard.cc (hunk 3 above)
Comment (maxz-lab) [reply net/socket/tcp_socket_starboard.cc:61]: Yes.

File: net/socket/tcp_socket_starboard.cc (hunk 4 above)
Comment (maxz-lab) [reply net/socket/tcp_socket_starboard.cc:73]: Structure SbSocket members are associated to each other, like the address, the type, the error, and socket_waiter. It is convenient to keep them in a group.

**Pull request closed without merging**